docs/proposals/templates.md

```diff
@@ -0,0 +1,350 @@
+potentially breaking assumed linkages to other pieces.  There is also no prescriptive way to define which configuration 
+values are appropriate for a deployer to tune or what the parameters control.
+
+## Use Cases
+
+### Use cases for templates in general
+
+* Providing a full baked application experience in a single portable object that can be repeatably deployed in different environments.
+  * eg Wordpress deployment with separate database pod/replica controller
+  * Complex service/replication controller/volume topologies
+* Bulk object creation
+* Provide a management mechanism for deleting/uninstalling an entire set of components related to a single deployed application
+* As a developer I want to select the template to use from a list of available templates
+* Enabling the creation of user interfaces that can guide an application deployer through the deployment process with descriptive help about the configuration value decisions they are making, and useful default values where appropriate
+* Exporting a set of objects in a namespace as a template so the topology can be inspected/visualized or recreated in another environment
+
+
+### Use cases for parameters within templates
+
+* Generate and share passwords between components (parameter value is provided to each component as an environment variable or as a Secret reference, with the Secret value being parameterized)
+* Allow for simple deployment-time customization of “app” configuration via environment values or api objects, e.g. memory 
+  tuning parameters to a MySQL image, Docker image registry prefix for image strings, pod resource requests and limits, default 
+  scale size.
+* Allow simple, declarative defaulting of parameter values and expose them to end users in an approachable way - a parameter 
+  like “MySQL table space” can be parameterized in images as an env var - the template parameters declare the parameter, give 
+  it a friendly name, give it a reasonable default, and informs the user what tuning options are available.
+* Customization of component names to avoid collisions and ensure matched labeling (eg replica selector value and pod label are 
+  user provided and in sync).
+* Customize cross-component references (eg user provides the name of a secret that already exists in their namespace, to use in 
+  a pod as a TLS cert).
+* Provide guidance to users for parameters such as default values, descriptions, and whether or not a particular parameter value 
+  is required or can be left blank.
+* Parameterize the replica count of a deployment or PetSet
+* Parameterize part of the labels and selector for a DaemonSet
+* Parameterize quota/limit values for a pod
+* Parameterize a secret value so a user can provide a custom password or other secret at deployment time
+
+
+## Design Assumptions
+
+The goal for this proposal is a simple schema which addresses a few basic challenges:
```

Comment:
I don't believe this proposal should try to establish a specific schema, language or format for templates. There will be many candidates and we should not try to mandate one, since we are not likely to address everyone's needs effectively.